### PR TITLE
Prepare and build projects with azure CI

### PR DESCRIPTION
**Pull request by mchindri**
.hex files will be added to the release directory

### CODE REVIEW

**Comment by mirceacaprioru** [comment]
This pr is huge pretty hard to follow. Be sure all ADICUP projects have semihosting disabled otherwise the hex files won't work. Also when building the Release configuration should be used.

File: build_projects.py [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+						to_blue(project)))
+
+def build_cces_project(adicup_location, project, project_dir, export_dir):
+	DEFAULT_WORKSPACE = os.path.join(adicup_location, 'workspace')
+	CCES_TEMPLATE = "cces -nosplash \
+-application com.analog.crosscore.headlesstools -data %s -project %s "
+	print_build("cces", project)
+
+	cces_cmd = CCES_TEMPLATE % (DEFAULT_WORKSPACE, project_dir)
+	run_cmd(cces_cmd + '-build Debug')
```
Comment (mirceacaprioru) [inline build_projects.py:61]: build Relase

File: build_projects.py [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+
+	binary = os.path.join(project_dir, 'Debug', project)
```
Comment (mirceacaprioru) [inline build_projects.py:63]: Release

**Comment by mirceacaprioru** [comment]
I would have split this into a PR to prepare the ground and the second one to implement the build pieces to much stuff in one PR. It is good practice to have PRs of 5 commits maximum if they have so much stuff to change.

File: build_projects.py (hunk 2 above)
Comment (mchindri) [reply build_projects.py:63]: No all projecte worked on release, that's why I choosed Debug

**Comment by mchindri** [comment]
v2:
 - Keep just CI commits.
 - Project preparation splited to: https://github.com/analogdevicesinc/EVAL-ADICUP3029/pull/72

File: build_projects.py (hunk 2 above)
Comment (adrimbarean) [reply build_projects.py:63]: The projects should all work in release and debug as long as optimization flags are the same (on the O0 level).

**Comment by mchindri** [comment]
v3:
 - Update build_projects.py to build Release

**Comment by mchindri** [comment]
v4: 
-  Rebase onto master

**Review by mirceacaprioru — APPROVED** [review]
